### PR TITLE
Copy runtime info in RIC fusion transformation

### PR DESCRIPTION
**Pull request by artkuli**
### Tickets:
 - *88597*

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ class Attribute {
         }
         auto gather = std::make_shared<opset8::Gather>(output, create_1d_const(order), create_1d_const({get_axis()}));
         input.replace_source_output(gather);
-        // TODO: copy runtime info from RIC sub-graph (ticket 88597)
+        ngraph::copy_runtime_info(nodes, gather);
```
Comment (ilyachur) [inline src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp:87]: Can we use `ov::copy_runtime_info()`?

**Review by ilyachur — COMMENTED** [review]
@itikhono please take a look

File: src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp [hunk 2]
```diff
@@ -190,14 +190,26 @@ void erase(T port) {
 }  // namespace ric_attr
 
 namespace init {
+
+namespace {
+
+void add_node_with_inputs_to_vector(const std::shared_ptr<ov::Node>& node, NodeVector& vector) {
+    vector.push_back(node);
+    const auto& inputs = node->inputs();
+    for (auto it = inputs.begin(); it != inputs.end(); ++it) {
```
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp:199]: Detail:
```suggestion
    for (const auto& input = inputs) {
        vector.push_back(input->get_source_output().get_node_shared_ptr());
    }
```
or use `for_each` ?

File: src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp (hunk 1 above)
Comment (artkuli) [reply src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp]: Changed 😊

File: src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp (hunk 2 above)
Comment (artkuli) [reply src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp]: Fixed 😊

**Review by praasz — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Pull request merged**